Microsoft.Azure.CosmosRepository/src/Providers/DefaultCosmosUniqueKeyPolicyProvider.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) IEvangelist. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Collections.Concurrent;
+using Microsoft.Azure.CosmosRepository.Attributes;
+using Microsoft.Azure.Cosmos;
+using System.Linq;
+using System.Reflection;
+
+namespace Microsoft.Azure.CosmosRepository.Providers
+{
+    class DefaultCosmosUniqueKeyPolicyProvider : ICosmosUniqueKeyPolicyProvider
+    {
+
+        static readonly Type _uniqueKeyAttributeType = typeof(UniqueKeyAttribute);
+        static readonly ConcurrentDictionary<Type, UniqueKeyPolicy> _uniqueKeyPolicyMap =
+            new ConcurrentDictionary<Type, UniqueKeyPolicy>();
+
+        /// <inheritdoc />
+        public UniqueKeyPolicy GetUniqueKeyPolicy<TItem>() where TItem : IItem =>
+            _uniqueKeyPolicyMap.GetOrAdd(typeof(TItem), GetPartitionKeyNameFactory);

```

Comment:
```suggestion
            _uniqueKeyPolicyMap.GetOrAdd(typeof(TItem), GetUniqueKeyPolicyFactory);
```

Name this more appropriately, as it was copied and pasted from the `DefaultCosmosPartitionKeyPathProvider`.